### DISCUSSION TITLE
CUE Tutorials - What would help you?

### DISCUSSION THREAD

**Discussion by mircubed** [dagger/dagger#2473] [General]
Hi Dagger Community! 

We want to make the onramp to CUE easier for the community, so I'd like to collect feedback on exactly where you might be getting stuck.

Currently, we point the community to this documentation: 
- Follow [CUE's tutorial ](https://cuetorials.com/)
- Get a quick understanding from Dagger's "[What is CUE?](https://docs.dagger.io/1215/what-is-cue/)" page 

**Please let us know where you have been getting stuck in your CUE journey, and what type of documentation or tutorial could have helped you.** 

Your feedback from this discussion will help us prioritize what CUE tutorials and doc improvements to do first. Thanks!

**Comment by martimors** [comment]
I think a CUE linter for VSCode, that highlights the parts with problems and a tooltip of what the error is, would go a long way. The next step would be some autocompletion based on the imports in the top of the file. This would significantly lower the barrier of getting started with CUE and Dagger in my opinion.

**Reply by dolanor** [reply]
FYI, we're currently working on a CUE LSP that should give all of that. :)

**Comment by gpoulin** [comment]
I think the CUE tutorial is relatively good. However what confused me at start is the relation between CUE and Dagger. A posteriori it looks obvious but it wasn't when I started. Knowing that I'm basically writing a JSON with some tooling (CUE) that then get interpreted by Dagger helped my mental model. To be honest, I'm still a little confused on how dagger generated fields work particularly when referencing them.
